### PR TITLE
Node16 docs update

### PR DESCRIPTION
**Pull request by gunamata**
Updated README.md and the windows-setup.ps1 for Windows development environment setup with node v16.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -109,11 +110,16 @@ You are now ready to clone the repository and run `npm install`.
...
+1. Install [Windows Subsystem for Linux (WSL)] on your machine. Skip this step, if WSL already installed.
 2. Install [Scoop] via `iwr -useb get.scoop.sh | iex`
 3. Install git, go, nvm, and unzip via `scoop install git go nvm unzip`
-4. Install NodeJS via `nvm install 14.17.0`
-  * Remember to use it by running `nvm use 14.17.0`
+4. Run `scoop bucket add versions`
+5. Install python via `scoop install python37`
+6. Install NodeJS via `nvm install 16.12.0`
+7. Install Visual Studio 2017 or higher. Make sure you have `Typescript SDK` and `Windows SDK` components installed.
+8. Ensure `msbuild_path` and `msvs_version` are configured correctly in `.npmrc` file.
```
Comment (mook-as) [inline README.md:120]: Are there instructions we can point people at, so they know what they need?

**Review by mook-as — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -109,11 +110,16 @@ You are now ready to clone the repository and run `npm install`.
...
 
-1. Install [Windows Subsystem for Linux (WSL)] on your machine.
+1. Install [Windows Subsystem for Linux (WSL)] on your machine. Skip this step, if WSL already installed.
 2. Install [Scoop] via `iwr -useb get.scoop.sh | iex`
 3. Install git, go, nvm, and unzip via `scoop install git go nvm unzip`
-4. Install NodeJS via `nvm install 14.17.0`
-  * Remember to use it by running `nvm use 14.17.0`
+4. Run `scoop bucket add versions`
+5. Install python via `scoop install python37`
+6. Install NodeJS via `nvm install 16.12.0`
```
[the remark this answers is not in the document]
Comment (rak-phillip) [reply README.md:118]: We should be able to run `nvm install 16` and nvm will fetch the latest release of v16. Similar to the python comment above.

File: README.md [hunk 3]
```diff
@@ -109,11 +110,16 @@ You are now ready to clone the repository and run `npm install`.
 
 ##### Manual Development Environment Setup
 
-1. Install [Windows Subsystem for Linux (WSL)] on your machine.
+1. Install [Windows Subsystem for Linux (WSL)] on your machine. Skip this step, if WSL already installed.
```
[the remark this answers is not in the document]
Comment (gunamata) [reply README.md:118]: I checked the links and they work for me. not sure what you are referring to .

File: README.md [hunk 4]
```diff
@@ -101,6 +101,12 @@ with an existing Windows installation.
    iwr -useb 'https://github.com/rancher-sandbox/rancher-desktop/raw/main/scripts/windows-setup.ps1' | iex
    ```
 4. Close the privileged PowerShell prompt.
+5. Ensure `msbuild_path` and `msvs_version` are configured correctly in `.npmrc` file. Run below commands to set these properties.
```
Comment (ericpromislow) [inline README.md:104]: I would write this as:

Ensure `msbuild_path` and `msvs_version` are configured correctly in `.npmrc`. Run the following commands to set these properties:

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ericpromislow) [reply README.md:118]: `if WSL is already installed.`

File: README.md [hunk 5]
```diff
@@ -109,13 +115,24 @@ You are now ready to clone the repository and run `npm install`.
...
-4. Install NodeJS via `nvm install 14.17.0`
-  * Remember to use it by running `nvm use 14.17.0`
+1. Install [Windows Subsystem for Linux (WSL)] on your machine. Skip this step, if WSL already installed.
+2. Open a privileged PowerShell prompt (hit Windows Key + `X` and open `Windows PowerShell (Admin)`).
+3. Install [Scoop] via `iwr -useb get.scoop.sh | iex`.
+4. Install git, go, nvm, and unzip via `scoop install git go nvm unzip`.
+   Check node version with `nvm list`. If node v16 is not set as the current version, set it as current using `nvm use 16.xx.xx`.
+5. Run `scoop bucket add versions`.
+6. Install python via `scoop install python`.
+7. Install Visual Studio 2017 or higher. Make sure you have `Windows SDK` component installed. This [Visual Studio docs] describes steps to
```
Comment (ericpromislow) [inline README.md:125]: "Make sure you have the `Windows SDK` component installed."

File: README.md [hunk 6]
```diff
@@ -109,13 +115,24 @@ You are now ready to clone the repository and run `npm install`.
...
+   install components.
+8. Ensure `msbuild_path` and `msvs_version` are configured correctly in `.npmrc` file. Run below commands to set these properties.
```
Comment (ericpromislow) [inline README.md:127]: Same requested change as above

**Review by ericpromislow — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (gunamata) [reply README.md:118]: Modified the statement as per the suggestion.

**Review by ericpromislow — APPROVED** [review]
LGTM

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mook-as) [reply README.md]: The `[Windows Subsystem for Linux (WSL)]` bit shows as plain text rather than a link.

File: README.md [hunk 7]
```diff
@@ -109,13 +115,24 @@ You are now ready to clone the repository and run `npm install`.
...
 ##### Manual Development Environment Setup
 
-1. Install [Windows Subsystem for Linux (WSL)] on your machine.
-2. Install [Scoop] via `iwr -useb get.scoop.sh | iex`
-3. Install git, go, nvm, and unzip via `scoop install git go nvm unzip`
-4. Install NodeJS via `nvm install 14.17.0`
-  * Remember to use it by running `nvm use 14.17.0`
+1. Install [Windows Subsystem for Linux (WSL)] on your machine. Skip this step, if WSL is already installed.
+2. Open a privileged PowerShell prompt (hit Windows Key + `X` and open `Windows PowerShell (Admin)`).
+3. Install [Scoop] via `iwr -useb get.scoop.sh | iex`.
```
Comment (mook-as) [inline README.md:120]: Scoop doesn't need to be privileged (and probably shouldn't be).  Not sure what we need it for (only thing should be nvm, but I thought that prompted itself).

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (mook-as) [reply README.md]: I wonder if I'm using a _different_ nvm or something; when I do `nvm install 16` it wants to install `16.0.0` instead.

File: scripts/windows-setup.ps1 [hunk 8]
```diff
@@ -48,13 +48,15 @@ if (!$SkipTools) {
 
         Invoke-WebRequest -UseBasicParsing -Uri 'https://get.scoop.sh' `
             | Invoke-Expression
+        scoop install git go nvm
         scoop bucket add versions
```
Comment (mook-as) [inline scripts/windows-setup.ps1:52]: We no longer need the `versions` bucket. (Also, we can combine the two `scoop install` lines.)

File: scripts/windows-setup.ps1 [hunk 9]
```diff
@@ -48,13 +48,15 @@ if (!$SkipTools) {
...
-        scoop install git go nvm python27
+        scoop install python
+        # Install and use latest node 16* version
+        nvm install 16
+        nvm use $(nvm list | Where-Object { $_ -match '16\..+\..+'} | Select-Object -First 1).Trim()
```
Comment (mook-as) [inline scripts/windows-setup.ps1:56]: … `nvm list | Select-String '[16\.[0-9.]+]' | Select-Object -First 1 | ForEach-Object { $_.Matches.Value }`
The `Select-String` part just simplifies the expression.
The `ForEach-Object` part ensures we don't get the "Currently using…" part, if it's active.

**Review by mook-as — CHANGES_REQUESTED** [review]

File: scripts/windows-setup.ps1 [hunk 10]
```diff
@@ -48,13 +48,15 @@ if (!$SkipTools) {
 
         Invoke-WebRequest -UseBasicParsing -Uri 'https://get.scoop.sh' `
             | Invoke-Expression
+        scoop install git go nvm
         scoop bucket add versions
-        scoop install git go nvm python27
+        scoop install python
+        # Install and use latest node 16* version
+        nvm install 16
```
Comment (rak-phillip) [inline scripts/windows-setup.ps1:55]: @mook-as raised a good point earlier, `nvm install 16` will install `16.0.0` when using newer versions of `nvm-windows`. It's probably best to explicitly specify the version here, unless there's a better approach that I'm not aware of.

We most recently tested on `16.13.1` while working on #1198.

File: README.md (hunk 7 above)
Comment (gunamata) [reply README.md:120]: `iwr` command doesn't work within normal `cmd` terminal (cmd.exe), so power shell is required to run the command `iwr -useb get.scoop.sh | iex`

File: scripts/windows-setup.ps1 (hunk 10 above)
Comment (gunamata) [reply scripts/windows-setup.ps1:55]: On my windows machine, I have nvm 1.1.9 which seems to be the latest nvm version. When I do `nvm install 16` it pulls the latest 16 version which is 16.13.2. Unless we want to restrict to the version (16.13.1) we tested, the provided instructions should work.

File: scripts/windows-setup.ps1 (hunk 9 above)
Comment (gunamata) [reply scripts/windows-setup.ps1]: @mook-as , Modified the command per your suggestion.

File: scripts/windows-setup.ps1 (hunk 8 above)
Comment (gunamata) [reply scripts/windows-setup.ps1]: Agreed.  Removed versions and combined the scoop install lines.

File: scripts/windows-setup.ps1 (hunk 10 above)
Comment (rak-phillip) [reply scripts/windows-setup.ps1:53]: I think it's good as long as it will install the latest version of node 16 instead of 16.0.0.

Comment (mook-as) [reply scripts/windows-setup.ps1:53]: Right, turns out the problem was on my side (I had nvm-windows 1.1.7); upgraded that and things work fine now.

File: README.md (hunk 7 above)
Comment (mook-as) [reply README.md:120]: Right, but that doesn't need to be an Admin PowerShell window. (just a normal, non-Admin one).

File: scripts/windows-setup.ps1 [hunk 11]
```diff
@@ -48,15 +48,14 @@ if (!$SkipTools) {
 
         Invoke-WebRequest -UseBasicParsing -Uri 'https://get.scoop.sh' `
             | Invoke-Expression
-        scoop bucket add versions
-        scoop install git go nvm python27
+        scoop install git go nvm python unzip
+        # Install and use latest node 16* version
+        nvm install 16
+        nvm list | Select-String '[16\.[0-9.]+]' | Select-Object -First 1 | ForEach-Object { $_.Matches.Value }
```
Comment (mook-as) [inline scripts/windows-setup.ps1:54]: Did you mean to run `nvm use` on the version number?

File: scripts/windows-setup.ps1 [hunk 12]
```diff
@@ -48,15 +48,14 @@ if (!$SkipTools) {
...
         # Temporarily commented out until we can handle later versions of node.js:
         # nvm install latest
         # nvm use $(nvm list | Where-Object { $_ } | Select-Object -First 1)
```
Comment (mook-as) [inline scripts/windows-setup.ps1:57]: This comment is probably not useful anymore

File: README.md [hunk 13]
```diff
@@ -101,6 +101,12 @@ with an existing Windows installation.
    iwr -useb 'https://github.com/rancher-sandbox/rancher-desktop/raw/main/scripts/windows-setup.ps1' | iex
    ```
 4. Close the privileged PowerShell prompt.
+5. Ensure `msbuild_path` and `msvs_version` are configured correctly in `.npmrc` file. Run the following commands to set these properties:
```
Comment (mook-as) [inline README.md:104]: Maybe add a new issue to do this automatically?

**Review by mook-as — CHANGES_REQUESTED** [review]

File: README.md (hunk 13 above)
Comment (gunamata) [reply README.md:104]: Agreed, added a new issue. https://github.com/rancher-sandbox/rancher-desktop/issues/1233

File: scripts/windows-setup.ps1 (hunk 12 above)
Comment (gunamata) [reply scripts/windows-setup.ps1]: yeah, removed the comment.

File: scripts/windows-setup.ps1 (hunk 11 above)
Comment (gunamata) [reply scripts/windows-setup.ps1]: oops, that was a mistake. I added `nvm use` to the command.

File: README.md [hunk 14]
```diff
@@ -92,15 +92,21 @@ with an existing Windows installation.
...
 1. Download a Microsoft Windows 10 [development virtual machine].
-2. Open a privileged PowerShell prompt (hit Windows Key + `X` and open
-   `Windows PowerShell (Admin)`).
+2. Open a PowerShell prompt (hit Windows Key + `X` and open
+   `Windows PowerShell`).
 3. Run the [automated setup script]:
    ```powershell
    Set-ExecutionPolicy RemoteSigned -Scope CurrentUser
 
    iwr -useb 'https://github.com/rancher-sandbox/rancher-desktop/raw/main/scripts/windows-setup.ps1' | iex
```
Comment (mook-as) [inline README.md:101]: The setup script tries to install Visual Studio components, which _does_ require admin.

**Review by mook-as — APPROVED** [review]

**Pull request merged**